client/src/components/login.tsx

```diff
@@ -0,0 +1,151 @@
+"use client"
+import { useState, useEffect } from 'react';
+
+import Image from 'next/image';
+import no_image from '../assets/no_image.png';
+
+const Login = () => {
+    const [isRegistration, setIsRegistration] = useState(false);
+    const [isValidating, setIsValidating] = useState(false);
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [forgotPassword, setForgotPassword] = useState(false);
+    const [passwordResetSent, setPasswordResetSent] = useState(false);
+
+    useEffect(() => {
```

Comment:
you have default values for this states why you need to set there state on 2 places?

Reply:
> you have default values for this states why you need to set there state on 2 places?

I don't know of these states.  where are they?  how are they implemented?

Reply:
In use effect

Reply:
I've revamped the whole series of screens, but the reason I do set the state in UseEffect, is because it's a top level component, which means that as a user flows through the screens, that state is being updated.  It never reverts to the default.  Using UseEffect ensures that every time I open the modal, that state is reverted to Login, rather than being any other state.